src/Nancy/ErrorHandling/DefaultStatusCodeHandler.cs

```diff
@@ -62,47 +66,60 @@ public void Handle(HttpStatusCode statusCode, NancyContext context)
                 return;
             }
 
-            string errorPage;
-
-            if (!this.errorPages.TryGetValue(statusCode, out errorPage))
+            if (!this.errorMessages.ContainsKey(statusCode) || !this.errorPages.ContainsKey(statusCode))
             {
                 return;
             }
 
-            if (String.IsNullOrEmpty(errorPage))
+            var result = new DefaultStatusCodeHandlerResult
             {
+                Details = StaticConfiguration.DisableErrorTraces ? DisableErrorTracesTrueMessage : context.GetExceptionDetails(),
+                Message = this.errorMessages[statusCode],
+                StatusCode = statusCode
+            };
+
+            try
+            {
+                context.Response = this.responseNegotiator.NegotiateResponse(result, context);
+                context.Response.StatusCode = statusCode;
                 return;
             }
-
-            Func<HttpStatusCode, NancyContext, string, string> expansionDelegate;
-            if (this.expansionDelegates.TryGetValue(statusCode, out expansionDelegate))
+            catch (ViewNotFoundException)
             {
-                errorPage = expansionDelegate.Invoke(statusCode, context, errorPage);
+                // No view will be found for `DefaultStatusCodeHandlerResult`
+                // because it is rendered from embedded resources below
             }
 
-            ModifyResponse(statusCode, context, errorPage);
+            this.ModifyResponse(statusCode, context, result);
         }
 
-        private static void ModifyResponse(HttpStatusCode statusCode, NancyContext context, string errorPage)
+        private void ModifyResponse(HttpStatusCode statusCode, NancyContext context, DefaultStatusCodeHandlerResult result)
         {
             if (context.Response == null)
             {
-                context.Response = new Response() { StatusCode = statusCode };
+                context.Response = new Response { StatusCode = statusCode };
             }
 
+            var contents = this.errorPages[statusCode];
+
+            if (!string.IsNullOrEmpty(contents))
+            {
+                contents = contents.Replace("[MESSAGE]", result.Message).Replace("[DETAILS]", result.Details);

```

Comment:
I could certainly do that instead. I was just extending the existing convention of replacing the `[DETAILS]` to keep the content consistent regardless of negotiation. I really have no strong opinions either way. Would you like to keep the message static and still dynamically render the details like the current implementation?